boot/efi_amd64.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
```

Comment:
it's a bit unclear to me why we are using architecture to control when we implement this, it's just reading files, there is not cpu arch related in this

Reply:
sure, I can remove that restriction, but I was unsure how we wanted to handle the decision of if we are on an EFI system or not

Reply:
we should find out what's the best way to check for that

Reply:
Dmitri says that looking for efivarfs being mounted is the best way to check right now.